src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,30 +37,33 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${selectedTodo.id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```

Comment:
I don't know whether this is needed and whether it won't be a mistake , but u can add aria-label to your button and delete this comment